### DISCUSSION TITLE
All text and symbols path in powershell is unreadable

### DISCUSSION THREAD

**Discussion by msp-incom** [JanDeDobbeleer/oh-my-posh#3936] [Troubleshoot] [answered]
I have installed oh-my-posh but cant figure out why my fonts (I presume) are not working.

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/130972246/23e03a9c-4477-44dd-992e-f55a87e97355)

My profile.ps1 only contains
`oh-my-posh init pwsh | Invoke-Expression`

I installed meslo using `oh-my-posh font install`

My terminal settings file is and I have installed MesloLGM Nerd Font (If its present in my font picker is it possible to have done this wrong somehow?)

`{
    "$help": "https://aka.ms/terminal-documentation",
    "$schema": "https://aka.ms/terminal-profiles-schema",
    "actions": [],
    "defaultProfile": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
    "profiles": 
    {
        "defaults": 
        {
            "font": 
            {
                "face": "MesloLGM Nerd Font"
            }
        },
        "list": 
        [
            {
                "font": 
                {
                    "face": "MesloLGM Nerd Font"
                },
                "guid": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
                "hidden": false,
                "name": "Windows PowerShell"
            },
            {
                "guid": "{0caa0dad-35be-5f56-a8ff-afceeeaa6101}",
                "hidden": false,
                "name": "Command Prompt"
            }
        ]
    }
}`

I have run `oh-my-posh init pwsh --config 'C:/Users/msp/ohmyposh.json' | Invoke-Expression w` with the following file

`{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "version": 2,
  "final_space": true,
  "blocks": [
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "type": "path",
          "style": "diamond",
          "powerline_symbol": "\ue0b0",
          "foreground": "#ffffff",
          "background": "#ff479c",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b0",
          "template": " \ue5ff {{ .Path }} ",
          "properties": {
            "style": "folder"
          }
        },
        {
          "type": "git",
          "style": "powerline",
          "powerline_symbol": "\ue0b0",
          "foreground": "#193549",
          "background": "#fffb38",
          "template": " {{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Staging.Changed) (.Working.Changed) }} |{{ end }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }}{{ end }} ",
          "properties": {
            "fetch_stash_count": true,
            "fetch_status": true,
            "fetch_upstream_icon": true
          }
        },
        {
          "type": "dotnet",
          "style": "powerline",
          "powerline_symbol": "\ue0b0",
          "foreground": "#ffffff",
          "background": "#6CA35E",
          "template": " \ue70c {{ if .Unsupported }}\uf071{{ else }}{{ .Full }}{{ end }} ",
          "properties": {
            "fetch_version": true
          }
        },
        {
          "type": "root",
          "style": "powerline",
          "powerline_symbol": "\ue0b0",
          "foreground": "#ffffff",
          "background": "#ffff66",
          "template": " \uf0e7 "
        },
        {
          "type": "exit",
          "style": "powerline",
          "powerline_symbol": "\ue0b0",
          "foreground": "#ffffff",
          "background": "#2e9599",
          "background_templates": [
            "{{ if gt .Code 0 }}#f1184c{{ end }}"
          ],
          "trailing_diamond": "\ue0b4",
          "template": " \ue23a{{ if gt .Code 0 }}\uf00d{{ else }}\uf42e{{ end }} ",
          "properties": {
            "always_enabled": true
          }
        }
      ]
    }
  ]
}
`

**Comment by JanDeDobbeleer** [comment]
@msp-incom this is due to encoding. If you can't change your computer's encoding, it is advised to change the encoding of PowerShell altogether. We do this also within oh-my-posh, but depending on the device configuration, it doesn't always work.

```
[Console]::OutputEncoding = [Text.Encoding]::UTF8
oh-my-posh init pwsh | Invoke-Expression
```

**Reply by msp-incom** [reply]
I would presume you mean I should add that to the profile, but when I do that I just get 
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/130972246/10c2a40f-0b1d-4e98-a2b0-2f1a510348d5)

**Reply by JanDeDobbeleer** [reply, marked as answer]
@msp-incom aha, you're running in constrained language mode. In that case you'll need to change this in the Windows settings (I have no idea where to be honest).

**Reply by msp-incom** [reply]
Thats too bad, can see it works if I open with administration rights, so year its probably my organization limiting attack vectors :/

But thanks for the guidance
